integrations/pjrt/requirements.txt

```diff
@@ -1,4 +1,4 @@
 -f https://iree.dev/pip-release-links.html
 -f https://storage.googleapis.com/jax-releases/jaxlib_nightly_releases.html
-iree-compiler==20230922.653
+iree-base-compiler==3.0.0
```

Comment:
FYI there is a warning (`ERROR: pip's dependency resolver`... ) in the logs: https://github.com/iree-org/iree/actions/runs/11986184740/job/33419112312?pr=19278#step:6:1117

```
Successfully built iree-pjrt-plugin-cpu
Installing collected packages: iree-pjrt-plugin-cpu
Successfully installed iree-pjrt-plugin-cpu-0.1.dev1

Notice:  A new release of pip is available: 24.0 -> 24.3.1
Notice:  To update, run: pip install --upgrade pip
Collecting jax==0.4.20
  Downloading jax-0.4.20-py3-none-any.whl.metadata (23 kB)
Collecting jaxlib==0.4.20
  Downloading jaxlib-0.4.20-cp311-cp311-manylinux2014_x86_64.whl.metadata (2.1 kB)
Collecting numpy<2
  Downloading numpy-1.26.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.0/61.0 kB 10.1 MB/s eta 0:00:00
Collecting ml-dtypes>=0.2.0 (from jax==0.4.20)
  Downloading ml_dtypes-0.5.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (21 kB)
Collecting opt-einsum (from jax==0.4.20)
  Downloading opt_einsum-3.4.0-py3-none-any.whl.metadata (6.3 kB)
Collecting scipy>=1.9 (from jax==0.4.20)
  Downloading scipy-1.14.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.8/60.8 kB 13.6 MB/s eta 0:00:00
Downloading jax-0.4.20-py3-none-any.whl (1.7 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 93.3 MB/s eta 0:00:00
Downloading jaxlib-0.4.20-cp311-cp311-manylinux2014_x86_64.whl (85.8 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 85.8/85.8 MB 43.0 MB/s eta 0:00:00
Downloading numpy-1.26.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.3 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18.3/18.3 MB 103.3 MB/s eta 0:00:00
Downloading ml_dtypes-0.5.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.5 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.5/4.5 MB 120.0 MB/s eta 0:00:00
Downloading scipy-1.14.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (41.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 41.2/41.2 MB 69.5 MB/s eta 0:00:00
Downloading opt_einsum-3.4.0-py3-none-any.whl (71 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 71.9/71.9 kB 18.6 MB/s eta 0:00:00
Installing collected packages: opt-einsum, numpy, scipy, ml-dtypes, jaxlib, jax
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
iree-base-compiler 3.1.0.dev0+1c1da1656545eccf02a1ee938800eaf16d49b969 requires sympy, which is not installed.
Successfully installed jax-0.4.20 jaxlib-0.4.20 ml-dtypes-0.5.0 numpy-1.26.4 opt-einsum-3.4.0 scipy-1.14.1
```

The `sympy` dep should not be needed for JAX / PJRT, so ignoring that should be fine. Context: https://github.com/iree-org/iree/pull/18262#discussion_r1722504803

---

Testing packages can also be tricky. Here's some background info that is tangentially related:

* For CI jobs that run on specific commits (`pull_request` and `push` triggers), we usually want to test the code at that commit, not an older version that was published somewhere (e.g. stable release from PyPI or nightly release from GitHub pages).
* For CI jobs that run on a schedule, (like [`samples.yml`](https://github.com/iree-org/iree/blob/main/.github/workflows/samples.yml), testing nightly or stable packages is more the norm.
* We expect that users and downstream developers will use published stable packages, though they may switch to nightly packages or source builds as needed.